primitives/weights/src/lib.rs

```diff
@@ -118,9 +118,75 @@ pub struct WeightToFeeCoefficient<Balance> {
 	pub degree: u8,
 }
 
-/// A list of coefficients that represent one polynomial.
+impl<Balance> WeightToFeeCoefficient<Balance>
+where
+	Balance: BaseArithmetic + From<u32> + Copy + Unsigned,
+{
+	/// Evaluate the term at `x` and saturatingly amalgamate into `result`.
+	///
+	/// The term is calculated as:
+	///
+	/// ```ignore
+	/// (frac * x^(degree) + integer * x^(degree)) * (-1 * c[0].negative)
```

Comment:
What is the numerical value of the `negative` field here? `true` maps to 1, but does `false` map to 0? If so, it would make the entire term 0, which is probably not what you want.